File: app/src/main/java/tpcreative/co/qrscanner/model/SyncDataModel.java
package tpcreative.co.qrscanner.model;
import com.google.gson.Gson;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;
import tpcreative.co.qrscanner.common.Utils;
import tpcreative.co.qrscanner.helper.SQLiteHelper;

public class SyncDataModel implements Serializable {
    public List<SaveModel> saveList;
    public List<HistoryModel> historyList;
    public String updatedDateTime;

    public SyncDataModel(boolean resultUpload){
        this.saveList = new ArrayList<>();
        for (SaveModel index : SQLiteHelper.getSaveList()){
            this.saveList.add(new SaveModel(index,true));
        }
        this.historyList = new ArrayList<>();
        for (HistoryModel index : SQLiteHelper.getHistoryList()){
            this.historyList.add(new HistoryModel(index,true));
        }
        this.updatedDateTime = Utils.getCurrentDateTimeSort();
        Utils.setLastTimeSynced(this.updatedDateTime);
        Utils.setRequestSync(false);
    }
    public String toJson(){
        return new Gson().toJson(this);
    }
}

File: app/src/main/java/tpcreative/co/qrscanner/model/SyncDataModel.kt
package tpcreative.co.qrscanner.model
import com.google.gson.Gson
import tpcreative.co.qrscanner.common.Utils
import tpcreative.co.qrscanner.helper.SQLiteHelper
import java.io.Serializable

class SyncDataModel(resultUpload: Boolean) : Serializable {
    var saveList: MutableList<SaveModel> = mutableListOf()
    var historyList: MutableList<HistoryModel> = mutableListOf()
    var updatedDateTime: String?
    fun toJson(): String? {
        return Gson().toJson(this)
    }

    init {
        for (index in SQLiteHelper.getSaveList()) {
            saveList.add(SaveModel(index, true))
        }
        for (index in SQLiteHelper.getHistoryList()) {
            historyList.add(HistoryModel(index, true))
        }
        updatedDateTime = Utils.getCurrentDateTimeSort()
        Utils.setLastTimeSynced(updatedDateTime)
        Utils.setRequestSync(false)
    }
}